### ISSUE TITLE
"link/iframe with identical name serve equivalent purpose" (b20e66/4b1c6c) have incorrect accessibility requirement mapping

### ISSUE THREAD

**Issue by Jym77** [act-rules/act-rules.github.io#968]
Following up on https://github.com/act-rules/act-rules.github.io/issues/962#issuecomment-539010202
```html
Joe is drinking a <a href=#cup-of-coffee>cup of joe</a>) in the 
<a href=#cup-of-a-guy-named-joe>cup of Joe</a>.
```
is an example of links with identical name that serve different purposes (thus failing rule [Links with identical accessible names have equivalent purpose](https://act-rules.github.io/rules/b20e66)) but the links are descriptive enough to pass [SC 2.4.9](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-link-only). 
Hence, the accessibility requirement mapping for this rule is wrong.

---
In the same line,
```
<iframe title="cup of joe" srcdoc="A cup of joe is slang for a cup of coffee" />

<iframe title="cup of joe" srcdoc="There's a guy named Joe who owns a cup. It's called the cup of Joe"/>
```
is an example of `iframe` with identical name that serve different purposes (thus failing rule [`iframe` elements with identical accessible names have equivalent purpose](https://act-rules.github.io/rules/4b1c6c)) but the names are descriptive enough to pass [SC 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value).
Hence, the accessibility requirement mapping for this rule is also wrong.

---
The rules might still map to another SC, or just be kept as a best practice…

**Comment by Jym77** [comment]
Thinking more about it, we get plenty of examples with all homonyms. "square" is an accurate description of both the geometric drawing and the arithmetic operation. Many people have identical names, and essentially the need for "disambiguation" pages on Wikipedia show that the same name can accurately describes two different things (the disambiguation pages themselves use context to make the purpose clear).

Note that context might not be enough, _i.e._ we might have links/iframe with same name and context, serving different purposes and still being descriptive enough. For example:
```html
The sailor <a href=https://en.wikipedia.org/wiki/John_Smith_(Medal_of_Honor,_born_1826)>
John Smith</a> received the medal of honor during the American Civil War.

The sailor <a href="https://en.wikipedia.org/wiki/John_Smith_(Medal_of_Honor,_born_1831)>
John Smith</a> received the medal of honor during the American Civil War.
```
So #693 has the same problem.

**Comment by Jym77** [comment]
Put assumption.

**Comment by Jym77** [comment]
Closed by #980.

**Comment by Jym77** [comment]
Unfortunately, I have to re-open this 😢 

[Understanding 2.4.4](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html) has a sentence saying:
> It is also a best practice for links with different purposes and destinations to have different link text.

Hence "identical links have identical purpose" is only a best practice for 2.4.4, not a requirement.
2.4.9 **does not** have the same remark. It still feels quite a stretch to assume that it is required for 2.4.9 when it's only a best practice for 2.4.4

I'd like to get other people's feedback on that…

**Comment by patrickhlauke** [comment]
Note that [Understanding 2.4.9](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-link-only) has, in its intent

> Best practice is that links with the same destination would have the same descriptions, but links with different purposes and destinations would have different descriptions

Also, speaking of iframes (per the title of this issue), xref https://github.com/w3c/wcag/issues/929 / https://github.com/act-rules/act-rules.github.io/issues/1008

**Comment by Jym77** [comment]
Damned, I missed that note in Understanding 2.4.9 😢 This is messing up with many other things…

**Comment by Jym77** [comment]
To sum up…

---
[Understanding 2.4.9](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-link-only.html) states:
> Best practice is that links with the same destination would have the same descriptions, but links with different purposes and destinations would have different descriptions

The second part of it, "links with different purposes and destinations would have different descriptions" is the implication "different purposes => different descriptions" which is logically equivalent to "same description => same purpose" that the published rule [Links with identical accessible names have equivalent purpose](https://act-rules.github.io/rules/b20e66) is checking.

Therefore, this rule is only checking a best practice and is not required for conformance to the SC. The rule currently has an assumption:
> This rule assumes that, within the context of the test subject, the description provided by the accessible name of a link can only accurately describe one resource (notably, homonyms alone are not used as link names).

but this is essentially assuming that a Best Practice is required for conformance.

Should we drop the accessibility requirement?

---
[Understanding 2.4.4](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html) states:
> It is also a best practice for links with different purposes and destinations to have different link text.

This is the implication "different purposes => different link text" which is logically equivalent to "same link text => same purpose" that the proposed rule #693 is checking.

Therefore, this rule is only checking a best practice and is not required for conformance to the SC.
but this is essentially assuming that a Best Practice is required for conformance.

Should we proceed anyway with the rule, without any accessibility requirement mapping but still being a very useful best practice? Or should we just drop it?

---
I'm less clear on the status for "iframe with identical name", but the discussion reported in #1008 (and 
https://github.com/w3c/wcag/issues/929) seems to indicate that is suffers from the same issue (not mapping to a SC).

**Comment by EmmaJP** [comment]
I can think of examples of different text linking to the same resource and not being an accessibility fail. However, if two links have the same text and link to different resources, then the link text should be changed to tell them apart.

**Comment by Jym77** [comment]
Decision on Nov 14 CG call: @WilcoFiers will escalate to AG. Leaving rule as it is for now.

**Comment by spanchang** [comment]
As suggested on the call: SC 2.4.4 (A) deals with link purpose in context while SC 2.4.9 (AAA) requires the link text to convey the purpose ... i.e. without reference to context in which the link appears.
Understanding SC 3.2.4(AA):
If there are two components on a web page that both have the same functionality as a component on another page in a set of web pages, then all 3 must be consistent. Hence the two on the same page will be consistent. 
While it is desirable and best practice always to be consistent within a single web page, 3.2.4 only addresses consistency within a set of web pages where something is repeated on more than one page in the set.
